src/Coroutine.php

```diff
@@ -65,7 +65,15 @@ public function __construct(callable $generatorFn)
                 $this->currentPromise->wait();
             }
         });
-        $this->nextCoroutine($this->generator->current());
+        try {
+            $this->nextCoroutine($this->generator->current());
+        }
+        catch (\Exception $exception) {
```

Comment:
this should be on the same line as the closing bracket. see coding style below in `_handleFailure`

Reply:
Sure, done